### ISSUE TITLE
testGrid reports error yet it still passes CircleCI

### ISSUE THREAD

**Issue by shaomeng** [NCAR/VAPOR#2764]
While manually execute the following command yields the following error, CircleCI passes.

./bin/testGrid -dims 80:80:80

```
*** Error reported in Regular grid ***

  Interpolation order: linear          
    RMS error:                                           0.0000
    Missing value count:                                 0
    GetValueAtIndex() vs GetValue() disagreement count:  0
    Time:                                                0.0000

*** Error reported in Regular grid ***

  Interpolation order: nearestNeighbor 
    RMS error:                                           0.0000
    Missing value count:                                 0
    GetValueAtIndex() vs GetValue() disagreement count:  0
    Time:                                                0.0000
```

**Comment by clyne** [comment]
@sgpearse there is a draft PR, #2767, that stops testGrid from erroneously reporting an error (yes there was an error in the error detection code). However, there is still a problem with the Python script used by CircelCI to invoke testGrid; it is not detecting the non-zero exit code that testGrid was generating. Leaving it up to use to fix the script.

**Comment by sgpearse** [comment]
Copied from the conversation in the PR #2767:

@shaomeng The smoke tests are passing on circleCI because we only test the grids listed below.  We can't test every permutation, so your 80x80x80 example indicates a new bug in the DataMgr.

`2x2x2
4x2x2
8x2x2
7x7x7
8x8x8`

**Comment by clyne** [comment]
@sgpearse per my comments above, the problematic permutation has been fixed in PR #2767. However, we still need to verify that if some other failure arrises, your python test driver is capturing the error code correctly. To verify, I suggest either testing the problematic permutation BEFORE #2767 is merged, or hardcoding testGrid to fail.

**Comment by sgpearse** [comment]
You are right John.  The python driver was not reporting an error code on grid test failure.  I updated #2767 to fix this.
